src/Core/Checkout/Cart/Rule/LineItemCustomFieldRule.php

```diff
@@ -83,6 +85,10 @@ public function getConstraints(): array
      */
     private function isCustomFieldValid(LineItem $lineItem): bool
     {
+        if (!$lineItem->hasPayloadValue('customFields')) {
+            return false;
+        }
+
         $customFields = $lineItem->getPayloadValue('customFields');
```

Comment:
`getPayloadValue()` already calls the `hasPayloadValue()` method, so you could just check if `$customFields` is null 🙂